clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/agent/AmazonApplicationLoadBalancerCachingAgent.groovy

```diff
@@ -158,6 +274,92 @@ class AmazonApplicationLoadBalancerCachingAgent extends AbstractAmazonLoadBalanc
       }
     }
 
+    // Get all the target groups
+    List<TargetGroup> allTargetGroups = []
+    DescribeTargetGroupsRequest describeTargetGroupsRequest = new DescribeTargetGroupsRequest()
+    while (true) {
+      def resp = loadBalancing.describeTargetGroups(describeTargetGroupsRequest)
+      allTargetGroups.addAll(resp.targetGroups)
+      if (resp.nextMarker) {
+        describeTargetGroupsRequest.withMarker(resp.nextMarker)
+      } else {
+        break
+      }
+    }
+
+    // Get all the target group health and attributes
+    Map<String, List<TargetHealthDescription>> targetGroupArnToHealths
+    Map<String, List<TargetGroupAttribute>> targetGroupArnToAttributes
+    if (account.eddaEnabled && eddaTimeoutConfig.albEnabled) {
+      List<TargetGroupAttributes> targetGroupAttributesList = eddaApi.targetGroupAttributes()
+      List<TargetGroupHealth> targetGroupHealthList = eddaApi.targetGroupHealth()
+      targetGroupArnToAttributes = targetGroupAttributesList.collectEntries { [(it.targetGroupArn): it.attributes] }
+      targetGroupArnToHealths = targetGroupHealthList.collectEntries { [(it.targetGroupArn): it.health]}
+    } else {
+      targetGroupArnToHealths = new HashMap<String, List<TargetHealthDescription>>()
+      targetGroupArnToAttributes = new HashMap<String, List<TargetGroupAttribute>>()
+      for (TargetGroup targetGroup : allTargetGroups) {
+        List<TargetHealthDescription> targetHealthDescriptions = loadBalancing.describeTargetHealth(
+          new DescribeTargetHealthRequest().withTargetGroupArn(targetGroup.targetGroupArn)
+        ).targetHealthDescriptions
+        targetGroupArnToHealths.put(targetGroup.targetGroupArn, targetHealthDescriptions)
+        List<TargetGroupAttribute> targetGroupAttributes = loadBalancing.describeTargetGroupAttributes(
+          new DescribeTargetGroupAttributesRequest().withTargetGroupArn(targetGroup.targetGroupArn)
+        ).attributes
+        targetGroupArnToAttributes.put(targetGroup.targetGroupArn, targetGroupAttributes)
+      }
+    }
+
+    Map<String, List<Listener>> loadBalancerArnToListeners = [:]
+    Map<Listener, List<Rule>> listenerToRules = new HashMap<>()
+    for (LoadBalancer lb : allLoadBalancers) {
+      // Add the listeners
+      List<Listener> listenerData = new ArrayList<>()
+      if (account.eddaEnabled && eddaTimeoutConfig.albEnabled) {
+        try {
+          listenerData = eddaApi.listeners(lb.loadBalancerName)
+        } catch (RetrofitError ignore) {
+          // this is acceptable since we may be waiting for the caches to catch up
```

Comment:
So edda could have the load balancer but then error when asking for it's listeners? Seems suspicious.